message-index/messages/GHC-99339/multiplemodules/index.md

```diff
@@ -0,0 +1,18 @@
+---
+title: Ambiguous record update (multiple modules)
+---
+
+Both records `R1` and `R2` share the same field `x`. For this reason, the compiler cannot infer a type for the record update `r { x = 1 }`.
+If the records are defined in separate modules, then the forward compatible way to solve this error is to use fully qualified field names.
```

Comment:
Note that the records don't necessarily need to be in separate modules, provided they are not both in the current module, because you can use `qualified` and restricted imports, e.g.
```hs
import qualified Module1 ( R1(x) ) as R1
```

Reply:
I tried to build a before/after example for this, but it seems this would trigger another error, not GHC-99339.
I.e. if I take the two modules:

```haskell
{-# LANGUAGE DuplicateRecordFields #-}
module Module1 where

data R1 = MkR1 { x :: Int }

data R2 = MkR2 { x :: Int }
```
and
```haskell
module Module2 where

import Module1

update r = r { x = 1 }
```
I get the following error message:
```
messages/GHC-99339/multiplemodules2/before/Module2.hs:5:16: error:
    Ambiguous occurrence ‘x’
    It could refer to
       either the field ‘x’ of record ‘R2’,
              imported from ‘Module1’ at messages/GHC-99339/multiplemodules2/before/Module2.hs:3:1-14
           or the field ‘x’ of record ‘R1’,
              imported from ‘Module1’ at messages/GHC-99339/multiplemodules2/before/Module2.hs:3:1-14
  |
5 | update r = r { x = 1 }
  |                ^
  ```
  
  This error apparently hasn't received a structured error message yet.

Reply:
I think you need `{-# LANGUAGE DuplicateRecordFields #-}` or at least `{-# LANGUAGE DisambiguateRecordFields #-}` on `Module2`? Otherwise indeed the ambiguity is reported sooner, because GHC doesn't even try to use type information to disambiguate.